src/catchup/CatchupWork.cpp

```diff
@@ -300,6 +321,40 @@ CatchupWork::runCatchupStep()
         CatchupRange{mLastClosedLedgerHashPair.first, resolvedConfiguration,
                      mApp.getHistoryManager()};
 
+    auto checkpoint = mApp.getHistoryManager().checkpointContainingLedger(
+        catchupRange.last());
+    auto ft =
+        FileTransferInfo(*mDownloadDir, HISTORY_FILE_TYPE_LEDGER, checkpoint);
+
+    if (!mDownloadStateForHerder)
+    {
+        mDownloadStateForHerder = addWork<GetAndUnzipRemoteFileWork>(ft);
+        return State::WORK_RUNNING;
+    }
+    else if (mDownloadStateForHerder->getState() !=
+             BasicWork::State::WORK_SUCCESS)
+    {
+        return mDownloadStateForHerder->getState();
+    }
+
+    if (mApp.getHerder().trackingConsensusLedgerIndex() < catchupRange.last())
+    {
+        auto entry = getHistoryEntryForLedger(catchupRange.last(), ft);
+        if (!entry)
+        {
+            CLOG_ERROR(History,
+                       "Malformed catchup file does not contain "
+                       "LedgerHeaderHistoryEntry for ledger {}",
+                       catchupRange.last());
+            return BasicWork::State::WORK_FAILURE;
+        }
+
+        mApp.getHerder().setTrackingSCPState(catchupRange.last(),
```

Comment:
Are you sure this is correct?

We only verify the ledger header (in particular if we use trusted hash) at step `// Step 4.1: Download and verify ledger chain` - so this seems to potentially call `setTrackingSCPState` with bad data?

I would imagine that we should call `setTrackingSCPState` as late as possible (but before we close ledgers).

Reply:
yes, you're totally right, this should be done later. Will fix.

Reply:
Moved this to after ledger verification success